app/game/game.js

```diff
@@ -85,7 +86,25 @@ const GAME = (humanUsername, aiUsername, gameId) => {
         }
 
         gameTime -= 1;
-
+        if (gameTime === 0) {
+            console.log("game-----------------------------------------------end");
+            // Check if both players are logged in
+            if (CLIENTS_HANDLER.areBothPlayersLoggedIn(gameId,AI_USERNAME,HUMAN_USERNAME)) {
+                // Send game end message to both players
+                sendDataToPlayer(gameId, HUMAN_USERNAME, GameEndData('You lost'));
+                sendDataToPlayer(gameId, AI_USERNAME, GameEndData('You lost'));
+                // Update player stats
+                CLIENTS_HANDLER.updatePlayerStats(HUMAN_USERNAME, 'You lost');
+                CLIENTS_HANDLER.updatePlayerStats(AI_USERNAME, 'You lost');
+                removeGame(gameId);
+            } else {
+                sendDataToPlayer(gameId, HUMAN_USERNAME, GameEndData('You lost'));
+                sendDataToPlayer(gameId, AI_USERNAME, GameEndData('You lost'));
+                console.log("At least one user is not loggedin");
+                removeGame(gameId);
+            }
+        }
```

Comment:
The `tick()` function is not a great place for this. When the time runs out, we could call an `onTimeOut()` function. If the ship explodes, we call `onShipExplode()`. After these functions notify the clients and update stats, both can call `endGame()` which removes the Game from GAMES and does any other needed clean-up.